### ISSUE TITLE
nose2 removes plugin command line options if they're enabled by config

### ISSUE THREAD

**Issue by OddBloke** [nose-devs/nose2#432]
This behaviour was confusing to me:

```
$ nose2 --pretty-assert

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
$ cat << EOF > unittest.cfg
[pretty-assert]
always-on = True
EOF
$ nose2 --pretty-assert
usage: nose2 [-s START_DIR] [-t TOP_LEVEL_DIRECTORY] [--config [CONFIG]]
             [--no-user-config] [--no-plugins] [--plugin PLUGINS]
             [--exclude-plugin EXCLUDE_PLUGINS] [--verbosity VERBOSITY]
             [--verbose] [--quiet] [--log-level LOG_LEVEL] [-B]
             [--coverage PATH] [--coverage-report TYPE]
             [--coverage-config FILE] [-C] [-D] [-F] [--log-capture] [-h]
             [testNames [testNames ...]]
nose2: error: Unrecognized arguments: --pretty-assert
$ echo $?
2
```

I would expect both invocations to behave exactly the same.

**Comment by sirosen** [comment]
This and the other issue you filed missed my inbox somehow. This seems to me to be "definitely wrong" and probably fairly tractable to fix when I get some time to look at it. I've queued it up for `0.9.1`.

**Comment by OddBloke** [comment]
No worries, thanks for picking them up now!

**Comment by sirosen** [comment]
I've tracked this down to the plugin metaclass. Seems the command-line switch isn't being added if `always-on` is detected: https://github.com/nose-devs/nose2/blob/01bbe475331fc3ea1f3b19dabe54245fe4fc5fb5/nose2/events.py#L43-L49

While I have tested an easy hack right there -- adding a `noop` callback in place of plugin registration in that case -- I want to think about this for a minute more, and maybe handle this in a different way.
Maybe make the registration step handle "oh, this plugin is already registered". (If we want to get buzzwordy about it, make it idempotent.)
